modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -142,20 +173,167 @@ MultiSmoothSuperellipsoidIC::computeSuperellipsoidCenters()
       }
 
       if (i == 0) rr = _range.norm();
+
+      //Check each new center candidate against all accepted ellisoids unless this is the first candidate
+      if (_prevent_overlap == 1 && rr > _bubspac && _centers.size() > 0)
+      {
+        _CentNum = _centers.size();
+        Real status = overlapCheck(newcenter, _as[_CentNum], _bs[_CentNum], _cs[_CentNum], _ns[_CentNum]);
+        //If candidate center fails overlapCheck ensure that rr is smaller than _bubspac to continue loop
+        if (status == 0)
+          rr = _bubspac - 1;
+      }
     }
 
     if (num_tries == _numtries)
       mooseError("Too many tries in MultiSuperellipsoidEllipseIC");
 
-    _centers[i] = newcenter;
+    _centers.push_back(newcenter);
   }
 }
 
 void
 MultiSmoothSuperellipsoidIC::computeSuperellipsoidExponents()
 {
-  _ns.resize(_numbub);
+  for (unsigned int el = 0; el < _numbub; ++el)
+    _ns.push_back(_exponent);
+}
+
+Real
+MultiSmoothSuperellipsoidIC::overlapCheck(const Point & newcenter, const Real & _nc_as, const Real & _nc_bs, const Real & _nc_cs, const Real & _nc_ns)
+{
+  Point l_p = newcenter;
+  Real _a = _nc_as;
```

Comment:
Don't declare function-scope variables with leading underscores, those names are reserved for class-protected variables.